cluster-autoscaler/cloudprovider/azure/azure_template.go

```diff
@@ -112,7 +112,11 @@ func buildNodeFromTemplate(scaleSetName string, template compute.VirtualMachineS
 	}
 	node.Status.Capacity[apiv1.ResourcePods] = *resource.NewQuantity(110, resource.DecimalSI)
 	node.Status.Capacity[apiv1.ResourceCPU] = *resource.NewQuantity(vmssType.VCPU, resource.DecimalSI)
-	node.Status.Capacity[gpu.ResourceNvidiaGPU] = *resource.NewQuantity(vmssType.GPU, resource.DecimalSI)
+	// isNPSeries returns if a SKU is an NP-series SKU
+	// SKU API reports GPUs for NP-series but it's actually FPGAs
+	if !isNPSeries(*template.Sku.Name) {
+		node.Status.Capacity[gpu.ResourceNvidiaGPU] = *resource.NewQuantity(vmssType.GPU, resource.DecimalSI)
```

Comment:
capacity can be set via VMSS tag overrides:

https://github.com/marwanad/autoscaler/blob/3f0155269c4d6fa12c53a4c5836adaec010f21c1/cluster-autoscaler/cloudprovider/azure/azure_template.go#L127-L130

No point in auto-populating device-plugin concerns.